Sources/Sentry/SentryTimeToDisplayTracker.m

```diff
@@ -0,0 +1,121 @@
+#import "SentryTimeToDisplayTracker.h"
+#import "SentryCurrentDate.h"
+#import "SentrySpan.h"
+#import "SentrySpanContext.h"
+#import "SentrySpanId.h"
+#import "SentrySpanOperations.h"
+#import "SentrySwift.h"
+
+#if SENTRY_HAS_UIKIT
+
+@interface
+SentryTimeToDisplayTracker ()
+
+@property (nonatomic, strong) NSDate *startDate;
+@property (nonatomic, strong) SentrySpan *initialDisplaySpan;
+@property (nonatomic, strong) NSDate *fullDisplay;
+@property (nonatomic, strong) NSString *controllerName;
+@property (nonatomic) BOOL waitFullDisplay;
+
+@end
+
+@implementation SentryTimeToDisplayTracker
+
+- (instancetype)initForController:(UIViewController *)controller
+               waitForFullDisplay:(BOOL)waitFullDisplay
+{
+    if (self = [super init]) {
+        self.startDate = [SentryCurrentDate date];
+        self.controllerName = [SwiftDescriptor getObjectClassName:controller];
+        self.waitFullDisplay = waitFullDisplay;
+    }
+    return self;
+}
+
+- (void)installForTracer:(SentryTracer *)tracer
+{
+    self.initialDisplaySpan =
+        [tracer startChildWithOperation:SentrySpanOperationUILoadInitialDisplay
+                            description:[NSString stringWithFormat:@"%@ initial display",
+                                                  self.controllerName]];
+}
+
+- (void)registerInitialDisplay
+{
+    if (self.initialDisplaySpan.timestamp != nil) {
+        return;
+    }
+
+    self.initialDisplaySpan.timestamp = [SentryCurrentDate date];
+
+    if (self.fullDisplay != nil &&
+        [self.fullDisplay compare:self.initialDisplaySpan.timestamp] == NSOrderedAscending) {
+        self.fullDisplay = self.initialDisplaySpan.timestamp;
```

Comment:
`m`: I don't understand why we must set the `fullDisplay` here.

Reply:
`reportFullyDisplayed` may be called before initial display happens.
It means the content is fully loaded, but the screen itself was not displayed yet. It will happen with initial display.